From b3df5bfc80bc9d63a083acd455b0765e813b1aed Mon Sep 17 00:00:00 2001
From: Caleb Connolly <caleb.connolly@linaro.org>
Date: Wed, 17 Apr 2024 12:27:51 +0100
Subject: [PATCH 090/142] HACK: arm64: dts: qcom: sdm845: disable qcrypto

The qcrypto driver has regressed on SDM845, any attempts to use it
result in an instant hard-crash and the device either resets or goes to
crashdump mode (depending on vendor configuration).

Disable it until this is resolved.

Signed-off-by: Caleb Connolly <caleb.connolly@linaro.org>
---
 arch/arm64/boot/dts/qcom/sdm845.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm845.dtsi b/arch/arm64/boot/dts/qcom/sdm845.dtsi
index 460d1b8537..1edcb8526f 100644
--- a/arch/arm64/boot/dts/qcom/sdm845.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845.dtsi
@@ -2735,6 +2735,11 @@ crypto: crypto@1dfa000 {
 				 <&apps_smmu 0x706 0x1>,
 				 <&apps_smmu 0x714 0x1>,
 				 <&apps_smmu 0x716 0x1>;
+
+			/* FIXME: disabled due to a regression causing
+			 * a trap to TZ and a hard reset if this device
+			 * is used. */
+			status = "disabled";
 		};
 
 		ipa: ipa@1e40000 {
-- 
2.49.0

